### PR TITLE
428 memory storage eav

### PR DESCRIPTION
**Pull request by AshantiMutinta**
Implemented Memory EAV. Only note is that you can't really make generic lambdas in rust so I was forced to create a bounded function for the filter but it turned out okay. Also moved a eav into a module because I thought it made sense.

### CODE REVIEW

File: core/src/eav/memory/mod.rs [hunk 1]
```diff
@@ -0,0 +1,212 @@
+use eav::{Attribute, Entity, EntityAttributeValue, EntityAttributeValueStorage, Value};
+use error::HolochainError;
+use std::collections::HashSet;
+
+pub struct MemoryStorageEav {
```
Comment (ddd-mtl) [inline core/src/eav/memory/mod.rs]: How about renaming this to `EavMemoryStorage` ?
that way we can rename the current `MemoryStorage` to `ContentMemoryStorage` and have consistent naming.

File: core/src/eav/memory/mod.rs [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    fn fetch_eav(
+        &self,
+        entity: Option<Entity>,
+        attribute: Option<Attribute>,
+        value: Option<Value>,
+    ) -> Result<HashSet<EntityAttributeValue>, HolochainError> {
+        Ok(self
+            .eavs
+            .iter()
+            .cloned()
```
Comment (lucksus) [inline core/src/eav/memory/mod.rs]: Is this needed?

File: core/src/eav/memory/mod.rs (hunk 1 above)
Comment (AshantiMutinta) [reply core/src/eav/memory/mod.rs]: That would be perfect! I will make those changes asap

File: core/src/eav/memory/mod.rs (hunk 2 above)
Comment (lucksus) [reply core/src/eav/memory/mod.rs]: Right, but doesn't the collect at the end create a new HashSet since it's not into_iter()?

Comment (AshantiMutinta) [reply core/src/eav/memory/mod.rs]: Yes absolutely but not of type EntityAttributeValue they would be of &EntityAttributeValue :(

**Pull request merged**